presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -333,7 +333,7 @@ private synchronized void notifyBlockedCallers()
         blockedCallers.clear();
         for (SettableFuture<?> blockedCaller : callers) {
             // Notify callers in a separate thread to avoid callbacks while holding a lock
-            executor.execute(() -> blockedCaller.set(null));
+            scheduler.execute(() -> blockedCaller.set(null));
```

Comment:
Do we really want a `ScheduledExecutorService` in `ExchangeClient`? Seems that we don't schedule anything on it, we just use it as an executor.